raiden/transfer/mediated_transfer/events.py

```diff
@@ -176,3 +184,30 @@ class EventRouteFailed(Event):
     secrethash: SecretHash
     route: List[Address]
     token_network_address: TokenNetworkAddress
+
+
+@dataclass
+class EventWrapper:
+    """
+    Some events have missing data and need to be encapsulated inside wrapping events that will
+    fetch this data before they can be processed.
+    """
+
+    events: List[Event]
+    _wrapping_map: Dict[object, Callable] = field(init=False)
+
+    def __post_init__(self) -> None:
+        self._wrapping_map: Dict[object, Callable] = defaultdict(lambda: lambda x: x)
+        self._wrapping_map.update({SendMessageEvent: self._wrap_send_message_event})
+
+    @staticmethod
+    def _wrap_send_message_event(event: SendMessageEvent) -> Event:
+        return RequestMetadata([event]) if not event.recipient_metadata else event
```

Comment:
How should we handle false metadata here? It could be the case that there are actually metadata but user_id key missing. Dont we already have a `verify_metadata` method?

Reply:
see [here](https://github.com/raiden-network/raiden/blob/cec446f9afa70778261d9fcb012930aa76cd786c/raiden/utils/validation.py#L19)

Reply:
We do, and we have this verification somewhere else. I don't think this is the right place to validate the data we have, only to check its existence.

Reply:
Ah okay, so you mean that we already checked it before?

Reply:
Yes, the classes `RouteState` and `RouteMetadata` inherit from. `MetadataValidation`, and both classes are used to implement the address to metadata dictionary.